[Feat] 로그인 로그아웃 기능 구현

yoonnable opened the pull request:
## 🌿 PR타입
- [x] 기능 추가
- [ ] 버그 수정
- [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트
- [ ] 기타 사소한 수정

## ❗️관련 이슈[close #3]
### 📑 개요
로그인, 로그아웃 기능을 구현합니다.
###  🧷 변경사항
<!--변경 내용을 적어주세요 (커밋 번호를 적어주세요)-->
- 81595f148a8121edbdf8de236540e9bba55a3d0f : 응답 메시지를 커스텀하였습니다.
- f453f9172234d904861884f6ade054643a1eb683 : 로그인 기능을 구현하였습니다.
  - session을 사용하였습니다.
- f4ad8b5310bdefcbbf0259e431072b0d2ad408c9 : 로그인 인증을 위해 인터셉터를 생성했습니다.
- 0c1597f91445ff879779ee3e6f3f8c3edca898a5 : 로그아웃을 구현하였습니다.

## 📷 스크린샷

## 👀 기타

yoonnable committed:
feat: 회원가입 기능 구현

yoonnable committed:
feat: create ResponseMessage

yoonnable committed:
feat: 로그인 기능 구현
- 세션 활용

yoonnable committed:
feat: 로그인 체크 인터셉터 구현

yoonnable committed:
feat: 로그아웃 구현

f-lab-jason commented on src/main/java/com/ecommerce/common/StatusEnum.java:
```diff
@@ -0,0 +1,17 @@
+package com.ecommerce.common;
+
+public enum StatusEnum {
```
스프링에서 제공하는 http response관련된 enum이 있을텐데 요걸 따로 선언하는것보단 그걸 그대로 쓰셔도 무방할 것 같아요. (HttpStatus 였던 것 같아요)

f-lab-jason commented on src/main/java/com/ecommerce/handler/AuthInterceptor.java:
```diff
@@ -0,0 +1,23 @@
+
+        Object loginMembersId = session.getAttribute("LOGIN_MEMBERS_ID");
+
+        if(loginMembersId == null) {
+            throw new IllegalArgumentException("로그인이 필요합니다.");
+
+        }
```
이전 PR에도 비슷한 구조가 있었던 것 같은데 google formatter 같은걸 적용해보시면 어떨까요?
사소한거지만 if문뒤에 한 칸을 띄운다거나 필요없는 빈 라인이 있다거나 하는걸 체크하는 용도로 사용해요.

f-lab-jason commented on src/main/java/com/ecommerce/handler/AuthInterceptor.java:
```diff
@@ -0,0 +1,23 @@
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+        HttpSession session = request.getSession();
+
+        Object loginMembersId = session.getAttribute("LOGIN_MEMBERS_ID");
```
nit: 로그인한 단일 사용자인데 member*s*인 이유가 있을까요?

f-lab-jason replied:
로그인한 유저를 가져오는것도 SessionUtil에서 제공해주는게 조금 더 자연스러울 것 같아요.

f-lab-jason commented on src/main/java/com/ecommerce/common/ResponseMessage.java:
```diff
@@ -0,0 +1,17 @@
+    private String message;
+    private Object data;
+
+    public ResponseMessage() {
+        this.status = StatusEnum.BAD_REQUEST;
+        this.message = null;
+        this.data = null;
+    }
```
이것 크게 필요없어보이는데 이 친구를 제거하고 builder패턴을 적용하는게 어떨까요?

f-lab-jason commented on src/main/java/com/ecommerce/member/controller/SigninController.java:
```diff
@@ -0,0 +1,31 @@
+
+    @GetMapping("/logout")
+    public void logout(HttpSession session) {
+        SessionUtil.logoutMemberId(session);
```
logout은 service에서 제공해주고 service가 이 SessionUtil의 메소드를 호출하는게 조금 더 자연스러워보이는데 어떻게 생각하시나요?

f-lab-jason commented on src/main/java/com/ecommerce/utils/SessionUtil.java:
```diff
@@ -0,0 +1,21 @@
+
+import jakarta.servlet.http.HttpSession;
+
+public class SessionUtil {
```
lombok에서 제공하는 `@UtilityClass`를 적용하면 좋을 것 같아요!

f-lab-jason reviewed (commented):
일단 로그인 자체는 인터셉터로 구현하시는건 괜찮지만 공부를 겸해서 spring security를 한 번 적용해보는것도 괜찮을 것 같아요 🙂

yoonnable committed:
style:: google format 적용

yoonnable committed:
fix: StatusEnum 삭제 -> HttpStatus 변경
fix: 로그인한 사용자 id ããkeyã값 수정
fix: ㄴsession 값 조회 시 SessionUtil 사용
fix: ResponseMessage builder 패턴 적용
fix: 로그아웃 비즈니스로ì§직 위치 수정
fix: @ãUtilityClass 적용
